### DISCUSSION TITLE
How to disable automatic system includes being appended?

### DISCUSSION THREAD

**Discussion by olliepop777** [microsoft/vscode-cmake-tools#3071] [Q&A] [answered]
It seems when I use the CMake Tools Extension it is automatically adding some system includes when I build. Is there any way to stop that behavior? When I build, I get stuff like this added in my output, leading to an error:

```
[build] In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/memory:671:
[build] In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/__functional_base:26:
[build] In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/utility:211:
[build] In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/__debug:14:
[build] In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/iosfwd:98:
[build] In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/__mbstate_t.h:29:
[build] In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/wchar.h:123:
[build] /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/wchar.h:89:10: fatal error: 'stdarg.h' file not found
```
I don't know why it's looking for that file in the wrong place/that doesn't exist. But I know that when I build outside of VSCode and CMake Tools, then I don't get these includes and errors. If I go in the `cmake-kits.json` and delete all the `compilers` settings completely, then these errors go away. I guess I am wondering if this is the right way to stop it from including those system includes (it doesn't seem right to me) and why the `compilers` setting is causing this to happen in the first place? It was just set by default like so:

```
"compilers": {
      "C": "/usr/bin/clang",
      "CXX": "/usr/bin/clang++"
    },
```

**Comment by gcampbell-msft** [comment, marked as answer]
@olliepop777 I think that this likely has to do with the selected kit that you are using. Additionally, you can use the "unspecified" kit, which doesn't use any of the cmake-kits.json and allows you to have full control over your kits setup.

Alternatively, you could integrate CMakePresets.json.
